src/global/project/environment.rs

```diff
@@ -0,0 +1,72 @@
+use std::{borrow::Borrow, fmt, str::FromStr};
+
+use miette::Diagnostic;
+use regex::Regex;
+use serde::{self, Deserialize, Deserializer};
+use thiserror::Error;
+
+/// Represents the name of an environment.
+#[derive(Debug, Clone, Eq, PartialEq, Hash)]
+pub(crate) struct EnvironmentName(String);
+
+impl EnvironmentName {
+    /// Returns the name of the environment.
+    pub fn as_str(&self) -> &str {
+        &self.0
+    }
+}
+
+impl Borrow<str> for EnvironmentName {
```

Comment:
also not sure if we need Borrow?

Reply:
Dunno. We have it here as well: https://github.com/prefix-dev/pixi/blob/e3b2fd8dcb2ad4665b6298330fdee98834a5f051/crates/pixi_manifest/src/environment.rs#L68

To me, it sounds reasonable, but I don't have strong feelings either way.

Reply:
I think it's best to remove it if we don't use it right now